app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx

```diff
@@ -63,21 +69,26 @@ export function Dialog(props: types.DialogProps) {
         {opts => (
           <>
             {typeof title === 'string' && (
-              <aria.Header className="center sticky flex flex-none border-b px-3.5 py-2.5 text-primary shadow">
-                <aria.Heading level={2} className="text-l my-0 font-semibold leading-6">
+              <aria.Header className="center sticky flex flex-none items-center border-b px-3.5 py-2.5 text-primary shadow">
+                <aria.Heading
+                  slot="title"
+                  level={2}
+                  className="text-l my-0 font-semibold leading-6"
+                >
                   {title}
                 </aria.Heading>
 
                 <ariaComponents.Button
                   variant="icon"
-                  className="my-auto ml-auto"
+                  className={clsx('my-auto ml-auto mr-[-4px]', { hidden: hideCloseButton })}
+                  size="custom"
                   onPress={opts.close}
                   icon={Dismiss}
                 />
               </aria.Header>
             )}
 
-            <div className="flex-1 shrink-0">
+            <div className="z-1 flex-1 shrink-0 p-3.5">
```

Comment:
do we actually need `z-1`?

Reply:
As far as I remember yes, because otherwise the content will be overlayed with ::before backdrop

Reply:
`relative` on the child should fix this. while i'm not against z-indexes, i think if we do end up needing them, we should be using named z-indexes for specific purposes, otherwise we just end up with arbitrary z-indexes as a way to brute-force the correct layering.

Reply:
I don't like z-indexes. Will try your solution :)